submissions/starter-code/scripts/articleView.js

```diff
@@ -0,0 +1,148 @@
+      }
+    }
+  });
+};
+
+articleView.handleAuthorFilter = function() {
+  $('#author-filter').on('change', function() {
+    if ($(this).val()) {
+      // TODO: If the select box was changed to an option that has a value, we need to:
+      //       1. Hide all the articles,
+      //       2. Fade in only the articles that match based on the author that was selected.
+      //          Use an "attribute selector" to find those articles that match the value,
+      //          and fade them in for the reader.
+      $('article').hide();
+      var filterValue = $(this).val();
+      $('article').each(function() {
+        var currentArticle = $(this);
+        if(currentArticle.attr('data-author') === filterValue) {
+          currentArticle.fadeIn();
+        }
+      });
+    } else {
+      // TODO: Otherwise, we should:
+      //       1. Show all the articles,
+      //       2. Except the one article we are using as a template.
+      $('article').not('.template').fadeIn();
+    }
+    // Reset the category-filter:
+    $('#category-filter').val('');
+  });
+};
+
+articleView.handleCategoryFilter = function() {
+  // TODO: Just like we do for #author-filter above, we should handle change
+  //       events on the #category-filter element. When an option with a value
+  //       is selected, hide all the articles, then reveal the matches.
+  //       When the blank (default) option is selected, show all the articles,
+  //       except for the template. Be sure to reset the #author-filter while you are at it!
+  $('#category-filter').on('change', function() {
+    if($(this).val()) {
+      $('article').hide();
+      var filterValue = $(this).val();
+      $('article').each(function() {
+        var currentArticle = $(this);
+        if(filterValue === currentArticle.attr('data-category')) {
+          currentArticle.fadeIn();
+        }
+      });
+    } else {
+      // TODO: Otherwise, we should:
+      //       1. Show all the articles,
+      //       2. Except the one article we are using as a template.
+      $('article').not('.template').fadeIn();
+    }
+    $('#author-filter').val('');
+  });
+};
+
+articleView.handleMainNav = function() {
+  // TODO: Add a delegated event handler to .main-nav element below that will
+  //        power the Tabs feature.
+  //       Clicking any .tab element should:
+  //        1. hide all the .tab-content sections.
+  //        2. fade in the single .tab-content section that is associated with
+  //            the clicked .tab element.
+  //         You may need to dynamically build a selector string (concatenation???)
+  //          with the correct ID, based on the data available to you on the .tab
+  //          element that was clicked.
+
+  $('.main-nav').on('click', '.tab', function(e) {
+    e.preventDefault();
+    $('.tab-content').hide();
+
+    var tabName = $(this).data('content');
+
+    $('.tab-content').each( function () {
+      if ($(this).is('#' + tabName)) {
```

Comment:
Do you need a loop here? Can't you just fade in the right tab instantly? You have a selector for it.